app/src/components/pages/searchPage/searchPage.tsx

```diff
@@ -49,14 +50,35 @@ const SearchPage = ({
   const pathname = usePathname();
   const headingRef = useRef<HTMLHeadingElement>(null);
   const isFirstLoad = useRef<boolean>(false);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [filtersExpanded, setFiltersExpanded] = useState(false);
 
   const updateURL = async (queryString: string) => {
     const newUrl = `${pathname}?${queryString}`;
     push(newUrl);
   };
 
   useEffect(() => {
-    if (isFirstLoad.current) {
+    setIsLoaded(true);
+    if (
+      searchManager.lastFilterRef?.current &&
+      (searchManager.filters.length > 0 || searchManager.sort)
+    ) {
+      // Search for the button or input associated with the last used filter/sort
+      const button = document.querySelector(
+        `button[aria-label="${searchManager.lastFilterRef?.current}"]`
```

Comment:
This works well. You're righ that it's brittle. Can you try it this with ids? I don't imagine the value name will change but if you set an id for each filter every time it's updated, it might make it more stable. B